app/lib/frontend/models.dart

```diff
@@ -239,6 +241,50 @@ abstract class SecretKey {
   ];
 }
 
+/// A random ID, typically sent via e-mail, to verify and confirm an action.
+///
+/// [action] identifies the trigger and scope of the verification. It ensures
+/// that the given verification instance can't be reused for different purpose
+/// (e.g. filling the [parameters] for one goal and then trigger an action in
+/// another part of the application for another goal).
+@db.Kind(name: 'Verification', idType: db.IdType.String)
+class Verification extends db.Model {
+  @db.DateTimeProperty()
+  DateTime created;
+
+  @db.DateTimeProperty()
+  DateTime expires;
+
+  @db.StringProperty(required: true)
+  String action;
+
+  @db.StringProperty()
```

Comment:
I'd expect the `action` (or some derivative of it) to be exposed as part of the URL. I think it adds a nice touch to the verification flow that the URL hints what action to expect when the recipient will click on it.

As it would be exposed and treated separately anyway, why not keep it separate?

Reply:
Mainly to simplify this class.
However I can see now how it might be useful.
What about the parameters? They could also be exposed in the URL - then it would be clear what package you are adding yourself to, and what the email is.

No strong opinions - I have not seen the client code yet.